synapse/python_dependencies.py

```diff
@@ -147,7 +147,10 @@ def check_requirements(for_feature=None):
             )
         except DistributionNotFound:
             deps_needed.append(dependency)
-            errors.append("Needed %s but it was not installed" % (dependency,))
+            errors.append(
+                "Needed %s for the '%s' feature but it was not installed"
+                % (dependency, for_feature)
```

Comment:
for_feature can be None, so this error would be nonsense

Reply:
I've made it predicated on for_feature with the old string as a backup. I still think it's valuable to include the feature name if it's available.

Reply:
non-blocking suggestion: I think this would be clearer as:

```python
if for_feature:
    errors.append(something)
else:
    errors.append(something_else)